src/caf/base/data_structures.py

```diff
@@ -2191,26 +2230,59 @@ def check_compatibility(targets: Collection[IpfTarget], adjust: bool = False):
             Whether to change the targets or just report on their compatibility.
         """
         targ_dict = {i: j for i, j in enumerate(targets)}
+        # add reference to end of targ_dict if exists
+        if reference is not None:
+            targ_dict[len(targets)] = reference
         rmses = {}
         for pos in list(itertools.combinations(reversed(targ_dict), 2)):
-            target_1, target_2 = targ_dict[pos[1]].data, targ_dict[pos[0]].data
-            if target_1.zoning_system != target_2.zoning_system:
+            skip = False
+            target_1, target_2 = targ_dict[pos[1]], targ_dict[pos[0]]
+            # target_2 has subsets, not appropriate for comparison
+            if len(target_2.segmentation.input.subsets) > 0:
                 continue
```

Comment:
There are a few times where rmses aren't produced, should we put np.nan or a warning to provide some additional information?